### PR TITLE
ADEN-5326 add outstream param

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/ADEN-5326

### CODE REVIEW

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 1]
```diff
@@ -106,8 +107,18 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
 		}
 	}
 
+	function getOutstreamData() {
+		var getVideoTargeting = prebid.get().getAdserverTargetingForAdUnitCode,
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: I'm not sure how this method is related to video. Can we rename this method?

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 2]
```diff
@@ -106,8 +107,18 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
...
+			videoTargeting = getVideoTargeting && getVideoTargeting('INCONTENT_PLAYER');
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: Is it ok that we limit only to this slot? Video is also displayed in TLM and MIC

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 3]
```diff
@@ -2,9 +2,10 @@
 define('ext.wikia.adEngine.slot.slotTargeting', [
 	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.utils.math',
+	'ext.wikia.adEngine.wrappers.prebid',
 	'wikia.abTest',
 	'wikia.instantGlobals'
-], function (adContext, math, abTest, instantGlobals) {
+], function (adContext, math, prebid, abTest, instantGlobals) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/436b1ba5bd4c7f390f86a7ee3026df2234425874/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/436b1ba5bd4c7f390f86a7ee3026df2234425874/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 4]
```diff
@@ -106,8 +111,19 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
 		}
 	}
 
+	function getOutstreamData() {
+        var context = adContext.getContext(),
+            getAdserverTargeting = prebid.get().getAdserverTargetingForAdUnitCode,
+			videoTargeting = getAdserverTargeting && getAdserverTargeting(videoSlots[context.targeting.skin]);
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: In Oasis `context.targeting.skin` should be always available: https://github.com/Wikia/app/blob/258cf92130916a8d3b6cf5e25177f1ed11c43b13/includes/Skin.php#L25

Mobile-wiki loads our [adsContext.js](https://github.com/Wikia/mobile-wiki/blob/542a1b700f35303e47fdd70ec74e82bb5c8895ca/app/modules/ads.js#L337) which depends on global ads object which in Oasis is set by the [AdContextService](https://github.com/Wikia/app/blob/dev/extensions/wikia/AdEngine/AdEngine2ContextService.class.php#L68) which uses the skin property linked above. But in mobile-wiki it's set by [Mercury API](https://github.com/Wikia/app/blob/147eb410f3127bc5112324f7fe7c79ec6471bd32/extensions/wikia/MercuryApi/MercuryApiController.class.php#L224) which means we're safe as well, aren't we? :japanese_ogre:

**Review by nandy-andy — APPROVED** [review]

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js (hunk 4 above)
Comment (kvas-damian) [reply extensions/wikia/AdEngine/js/slot/slotTargeting.js]: Please fix indentation :)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/80ec4f3edad94a15c09bf472fcae8811c708e2d0/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/80ec4f3edad94a15c09bf472fcae8811c708e2d0/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 5]
```diff
@@ -106,8 +112,27 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
 		}
 	}
 
+	function getOutstreamData() {
+        var bidderName,
+			context = adContext.getContext(),
+            getAdserverTargeting = prebid.get().getAdserverTargetingForAdUnitCode,
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: Some indentations problem here :( it looks like spaces, not tabs

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 6]
```diff
@@ -106,8 +112,27 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
...
+            isVideoProvider,
+			videoTargeting = getAdserverTargeting && getAdserverTargeting(videoSlots[context.targeting.skin]);
+
+		if (videoTargeting) {
+			bidderName = videoTargeting.hb_bidder;
+            isVideoProvider = videoProviders.indexOf(bidderName) > -1;
+
+            if (isVideoProvider && videoTargeting.hb_pb) {
+				return bidderName.substr(0, 2) + math.leftPad(parseFloat(videoTargeting.hb_pb) * 100, 4);
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js]: Try to hide that behind some nice looking function

File: extensions/wikia/AdEngine/js/spec/slot/slotTargeting.spec.js [hunk 7]
```diff
@@ -217,4 +231,60 @@ describe('ext.wikia.adEngine.slot.slotTargeting', function () {
...
+					outstream: 'vu0000'
+				},
+				{
+					targeting: {
+						hb_bidder: 'vulcan',
+						hb_pb: '12.50'
+					},
+					outstream: 'vu1250'
+				},
+                {
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/spec/slot/slotTargeting.spec.js]: Spaces here, again :(

**Review by v3rth — CHANGES_REQUESTED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/46ac91cdf18ea3f94037f1ad85041b5639efe125/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/46ac91cdf18ea3f94037f1ad85041b5639efe125/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/89b1aad1732f012a346650fc707e9586e485ab46/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/89b1aad1732f012a346650fc707e9586e485ab46/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/502298bae37a21d1f1103f8b817ec887dd3cb79c/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/502298bae37a21d1f1103f8b817ec887dd3cb79c/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by v3rth — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/2ae74914493987010bc756ce95e24025d46d658b/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/2ae74914493987010bc756ce95e24025d46d658b/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by v3rth — APPROVED** [review]

File: extensions/wikia/AdEngine/js/slot/slotTargeting.js [hunk 8]
```diff
@@ -3,8 +3,9 @@ define('ext.wikia.adEngine.slot.slotTargeting', [
 	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.utils.math',
 	'wikia.abTest',
-	'wikia.instantGlobals'
-], function (adContext, math, abTest, instantGlobals) {
+	'wikia.instantGlobals',
+	require.optional('ext.wikia.adEngine.wrappers.prebid')
+], function (adContext, math, abTest, instantGlobals, prebid) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/slot/slotTargeting.js:8]: MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/19b8930c8980f8c7d9f7d61a7413301b58e725c2/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/19b8930c8980f8c7d9f7d61a7413301b58e725c2/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/5d88a2bde83bb772f91ca3c16a17cbec227cb3e5/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/5d88a2bde83bb772f91ca3c16a17cbec227cb3e5/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [helper.js#L19](https://github.com/Wikia/app/blob/f53dd81a2485829d5f2553eb27347c429439b7b0/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L19): Function has 15 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [helper.js#L59](https://github.com/Wikia/app/blob/f53dd81a2485829d5f2553eb27347c429439b7b0/extensions/wikia/AdEngine/js/provider/gpt/helper.js#L59): Function has a complexity of 16 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by bulaszenko — APPROVED** [review]

**Pull request merged**